### ISSUE TITLE
ContextManager: STATE_PROVIDER_TIMEDOUT  【DeadLock - 2】

### ISSUE THREAD

**Issue by xlb767923274** [alexa/avs-device-sdk#1977]
**IMPORTANT**: Before you create an issue, please take a look at our [Issue Reporting Guide](https://github.com/alexa/avs-device-sdk/wiki/Issue-Reporting-Guide).

## Briefly summarize your issue:

Repeatedly play music and joke, after a period of time, STATE_PROVIDER_TIMEDOUT error appears，and alexa can wake up no longer。

## What is the expected behavior?

STATE_PROVIDER_TIMEDOUT does not appear, alexa can wake up

## What behavior are you observing?

STATE_PROVIDER_TIMEDOUT error appears，and alexa can wake up no longer。

## Provide the steps to reproduce the issue, if applicable:

1. alexa, play music,
2. Interval 20s
3. alexa, tell me a joke
4. Interval 20s
5. Repeat 1-4

### What version of the AVS Device SDK are you using?
      <x.y.z>avs sdk 1.24.0 + ss sdk 2.7.0 + apl core1.6.1

### Tell us what hardware you're using:
- [ ] Desktop / Laptop
- [ ] Raspberry Pi
- [* ] Other - tell us more: RK3326

### Tell us about your OS (Type & version):
- [ ] Linux
- [ ] MacOS 
- [ ] Raspbian Stretch
- [ ] Raspbian Jessy
- [* ] Other - tell us more:Android8.1 & Android10

Logfile  in debug9 ，The file is too big, I only kept the problematic part

**Comment by xlb767923274** [comment]
because of #1965 ，I temporarily modified AndroidSLESMediaPlayer::configureNewRequest()

=====》

    stop(m_sourceId); //move stop to here, fix dead lock
    std::lock_guard<std::mutex> requestLock{m_requestMutex};
    {
        // Use global lock to stop player and set new source id.
        std::lock_guard<std::mutex> lock{m_operationMutex};
        if (m_hasShutdown) {
            return ERROR;
        }
        //stopLocked();
        m_sourceId = g_id.fetch_add(1);
        m_almostDone = false;
        m_initialOffset = offset;
    }
    ......
    m_mediaQueue.reset();  // Delete old queue before configuring new one.    
    ACSDK_DEBUG9(LX(__func__).d("to", "FFmpegDecoder::create()"));

**Comment by xlb767923274** [comment]
There is still a deadlock problem here after I modify it。。

Thread1 - AudioPlayer::configureMediaPlayer()
    ->AndroidSLESMediaPlayer::configureNewRequest() ,     
         {
               std::lock_guard<std::mutex> requestLock{m_requestMutex};     
               m_mediaQueue.reset();  // Delete old queue before configuring new one.
         }
     The reset operation is also within the scope of the lock；
     m_mediaQueue.reset() will to destruct m_mediaQueue， 
```

AndroidSLESMediaQueue::~AndroidSLESMediaQueue() {
    。。。
    m_decoder->abort();

    m_executor.waitForSubmittedTasks();
    m_executor.shutdown();
    。。。
}
```
Here must wait for all tasks in m_executor to end before returning; 
All tasks include AndroidSLESMediaQueue::fillbuffer()，
Maybe at the same time, AndroidSLESMediaQueue::fillbuffer() triggered a m_eventCallback(), 
       ->AndroidSLESMediaPlayer::onQueueEvent(） //here maybe request the same lock 'm_requestMutex' in thread2


in other words, 

thread1 locked m_requestMutex and wait thread 2 exit
thread2 maybe request to lock the same m_requestMutex before exit

**Comment by xlb767923274** [comment]
I don't know how to fix it, can you give me some suggestions before the next release version is released?

**Comment by blazusiak-amazon** [comment]
Hello @xlb767923274.

Are you still experiencing this issue on the latest version of the SDK (3.0)?
